indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/BERTags.kt

```diff
@@ -47,29 +50,30 @@ object BERTags {
     const val CONSTRUCTED: UByte = 0x20u // decimal 32
     const val UNIVERSAL: UByte = 0x00u // decimal 32
     const val APPLICATION: UByte = 0x40u // decimal 64
-    const val TAGGED: UByte = 0x80u // decimal 128 - maybe should deprecate this.
     const val CONTEXT_SPECIFIC: UByte = 0x80u // decimal 128
     const val PRIVATE: UByte = 0xC0u // decimal 192
     const val FLAGS: UByte = 0xE0u
 
 }
 
-object DERTags {
-    val DER_SEQUENCE: UByte = BERTags.CONSTRUCTED or BERTags.SEQUENCE
-    val DER_SET: UByte = BERTags.CONSTRUCTED or BERTags.SET
-    fun UByte.toExplicitTag() = BERTags.CONSTRUCTED or BERTags.TAGGED or this
-
-    val UByte.isExplicitTag get() = ((this and BERTags.CONSTRUCTED) != 0.toUByte()) && ((this and BERTags.TAGGED) != 0.toUByte())
+/**
+ * Convenience helper to easily construct implicitly tagged elements. Can be CONSTRUCTED or PRIMITIVE
+ * @param constructed whether to set the constructed bit
+ */
+fun ULong.toImplicitTag(constructed: Boolean = false) =
+    Asn1Element.Tag(this, constructed = constructed, tagClass = TagClass.CONTEXT_SPECIFIC)
 
-    fun UInt.toExplicitTag() = toUByte().toExplicitTag()
+fun UByte.isConstructed() = this and BERTags.CONSTRUCTED != 0.toUByte()
 
-    @Throws(Asn1Exception::class)
-    fun UInt.toImplicitTag() = runRethrowing { toUByte().toImplicitTag() }
 
-    @Throws(Asn1Exception::class)
-    fun UByte.toImplicitTag() = runRethrowing {
-        if (isContainer()) throw IllegalArgumentException("Implicit tag $this would result in CONSTRUCTED bit set") else BERTags.TAGGED or this
+enum class TagClass(val byteValue:UByte) {
```

Comment:
Please reformat automatically

Reply:
latest IDEA seems to be pretty much broken in that respect for this project